### PR TITLE
(WIP)4.0.8 single click payment

### PR DESCRIPTION
**Pull request by margud**

### CODE REVIEW

File: controllers/front/payment.php [hunk 1]
```diff
@@ -246,6 +247,16 @@ protected function createPayment($data, $selectedApi)
...
     }
 
+    protected function createCustomer($data, $selectedApi)
+    {
+        try {
+            $payment = $this->module->api->customers->create($data, array('embed' => 'payments'));
+        } catch (Exception $e) {
+            throw new ApiException($e->getMessage());
+        }
+        return $payment;
```
Comment (aleksiuno) [inline controllers/front/payment.php:257]: New line before return

File: src/Exception/MollieException.php [hunk 2]
```diff
@@ -4,4 +4,5 @@
 
 class MollieException extends \Exception
 {
+    const CUSTOMER_EXCEPTION = 0;
```
Comment (tomas862) [inline src/Exception/MollieException.php:7]: an exception which has **0** is the default one - all custom exceptions should start from 1

File: src/DTO/Line.php [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    /**
+     * @var string
+     */
+    private $vatRate;
+
+    /**
+     * @var Amount
+     */
+    private $unitPrice;
+    /**
```
Comment (aleksiuno) [inline src/DTO/Line.php:54]: Missing new lines before instance variables here and bellow

File: src/DTO/Line.php [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+     * @return mixed
+     */
+    public function getProductUrl()
+    {
+        return $this->productUrl;
+    }
+
+    /**
+     * @param mixed $productUrl
```
Comment (aleksiuno) [inline src/DTO/Line.php:124]: It's always a good idea to typehint a concrete types instead of mixed

File: src/DTO/Line.php [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+                    "value" => $this->getDiscountAmount()->getValue()
+                ]
+                :
+                [],
+            "vatAmount" => [
+                "currency" => $this->getVatAmount()->getCurrency(),
+                "value" => $this->getVatAmount()->getValue()
+            ]
+        ];
+    }
```
Comment (aleksiuno) [inline src/DTO/Line.php:289]: No new line at the end of the file

File: src/DTO/Object/Amount.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * @param float $value
+     */
+    public function setValue($value)
+    {
+        $this->value = $value;
+    }
+
+}
```
Comment (aleksiuno) [inline src/DTO/Object/Amount.php:88]: No new lines all around, you can config your IDE to add it automatically

File: src/DTO/OrderData.php [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+    /**
+     * @var
+     */
+    private $webhookUrl;
+
+    /**
+     * @var string
+     */
+    private $method;
+    /**
```
Comment (aleksiuno) [inline src/DTO/OrderData.php:65]: New lines before new instance variables

File: src/DTO/Line.php [hunk 8]
```diff
@@ -0,0 +1,290 @@
+<?php
```
Comment (aleksiuno) [inline src/DTO/Line.php:1]: Some classes have a license some not.

File: src/Utility/TextFormatUtility.php [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Mollie\Utility;
+
+class TextFormatUtility
+{
+    public static function formatNumber($unitPrice, $apiRoundingPrecision, $docPoint = '.', $thousandSep = '')
+    {
+        return number_format($unitPrice, $apiRoundingPrecision, $docPoint, $thousandSep);
+    }
+
```
Comment (aleksiuno) [inline src/Utility/TextFormatUtility.php:11]: Empty line is not necessary

**Pull request merged**